addons/survey/tests/test_survey_invite.py

```diff
@@ -258,3 +258,51 @@ def test_survey_invite_token_by_email_nosignup(self):
         answers = self.env['survey.user_input'].search([('survey_id', '=', self.survey.id)])
         self.assertEqual(len(answers), 1)
         self.assertEqual(answers.partner_id.display_name, first_partner.display_name)
+
+    @users('survey_user')
+    def test_survey_invite_with_template_attachment(self):
+        """
+        Test that a group_survey_user can send a survey that includes an attachment from the survey invite's
+            email template
+        """
+        mail_template = self.env['mail.template'].create({
+            'name': 'test mail template',
+            'attachment_ids': [Command.create({
+                'name': 'some_attachment.pdf',
+                'res_model': 'mail.template',
+                'datas': 'test',
+                'type': 'binary',
+            })],
+        })
+
+        self.user_survey = self.env['survey.survey'].with_user(self.survey_user).create({
```

Comment:
As the test has a decorator, no need to use with_user (and no need to store on self, we can use internal variables), like
```suggestion
        user_survey = self.env['survey.survey'].create({
```